src/pushprocessor.ts

```diff
@@ -162,24 +168,31 @@ const EXPECTED_DEFAULT_UNDERRIDE_RULE_IDS = [
 /**
  * Make sure that each of the rules listed in `defaultRuleIds` is listed in the given set of push rules.
  *
+ * @param kind - the kind of push rule set being merged.
  * @param incomingRules - the existing set of known push rules for the user.
  * @param defaultRules - a lookup table for the default definitions of push rules.
  * @param defaultRuleIds - the IDs of the expected default push rules, in order.
  *
  * @returns A copy of `incomingRules`, with any missing default rules inserted in the right place.
  */
 function mergeRulesWithDefaults(
+    kind: PushRuleKind,
```

Comment:
this is unused?

Reply:
If it was unused the linter would throw a tantrum, it is used for the logging

Reply:
I mean, fair, but also: what logging? It's entirely possible I'm blind, but I really can't see it?

Reply:
![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/bfd7df62-58db-4012-a14b-3d938caea984)

Reply:
Huh, somehow the `${kind}` went missing in my iterations, good spot